python_katas/kata_1/test.py

```diff
@@ -69,78 +111,190 @@ class TestIsUniqueString(unittest.TestCase):
     """
 
     def test_sample(self):
-        # your code here
-        pass
+        some_str = ('MynameIsmMax')
+        self.assertEqual(questions.is_unique_string(some_str), False)
+        some_str = ('aNuke')
+        self.assertEqual(questions.is_unique_string(some_str), True)
+        some_str = ('IAmDeath')
+        self.assertEqual(questions.is_unique_string(some_str), True)
+
+
 
 
 class TestListDiff(unittest.TestCase):
     """
     1 Katas
     """
-
     def test_sample(self):
-        # your code here
-        pass
+        test_list = [1, 5, 0, 4, 1, 1, 1]
+        expected = [None, 4, -5, 4, -3, 0, 0]
+        result = questions.list_diff(test_list)
+        self.assertEqual(result, expected)
+
+    def test_empty(self):
+        test_list = []
+        expected = []
+        self.assertEqual(questions.list_diff(test_list), expected)
 
+    def test_one(self):
+        test_list = [1]
+        expected = [None]
+        self.assertEqual(questions.list_diff(test_list), expected)
 
 class TestPrimeNumber(unittest.TestCase):
     """
     1 Katas
     """
 
-    def test_sample(self):
-        # your code here
-        pass
+    def test_prime_1(self):
+        number = 7
+        self.assertEqual(questions.prime_number(number), True)
+
+    def test_prime_2(self):
+        number = 17
+        self.assertEqual(questions.prime_number(number), True)
+
+    def test_prime_3(self):
+        number = 13
+        self.assertEqual(questions.prime_number(number), True)
+
+    def test_not_prime_1(self):
+        number = 1
+        self.assertEqual(questions.prime_number(number), False)
+
+    def test_not_prime_2(self):
+        number = -20
+        self.assertEqual(questions.prime_number(number), False)
+
+    def test_not_prime_3(self):
+        number = 2.4
+        self.assertEqual(questions.prime_number(number), False)
 
 
 class TestPalindromeNum(unittest.TestCase):
     """
     1 Katas
     """
+    def test_pal1True(self):
+        num = 1441
+        self.assertTrue(questions.palindrome_num(num), "the num is palindrome but you return False")
 
-    def test_sample(self):
-        # your code here
-        pass
+    def test_pal1True(self):
```

Comment:
this function will overwrite the function defined in line 178 as they have exactly the same name